cpp/src/arrow/compute/kernels/vector_sort_test.cc

```diff
@@ -45,6 +45,15 @@ using internal::checked_pointer_cast;
 
 namespace compute {
 
+#ifdef ARROW_VALGRIND
+using RealArrowTypes = ::testing::Types<FloatType>;
```

Comment:
Ah, I didn't know you could redefine an alias with `using`.
As a style note, I would personally prefer if this created a distinct alias, to avoid confusion - and potential leaking accross unity builds?
(e.g. replace `RealArrowTypes` with `RealSortTypes` or something)